hawkular-inventory-impl-cassandra/src/main/java/org/hawkular/inventory/impl/cassandra/CassandraBackend.java

```diff
@@ -159,8 +170,59 @@ public CassandraBackend(RxSession session) {
     @Override
     public Iterator<Row> getTransitiveClosureOver(Row startingPoint, Relationships.Direction direction,
                                                        String... relationshipNames) {
-        //TODO implement
-        return null;
+        String startingCP = startingPoint.getString(Statements.CP);
+        if (startingCP == null) {
+            // Not a vertex
+            return Collections.<Row>emptyList().iterator();
+        }
+        Observable<Row> closureContent = getRelationshipTables(direction)
+                .flatMap(table -> {
+                    Statement q = select().all().from(table).where(in(Statements.NAME, Arrays.asList(relationshipNames)));
```

Comment:
Given what we said yesterday, would it make sense to add a feedId in the primary keys?